# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/windows/nsWindow.cpp
# Commit: 2424d5c30267
# Full Hash: 2424d5c30267f67ab2b28119d75d89f0441b30e0
# Author: Gijs Kruitbosch <gijskruitbosch@gmail.com>
# Date: 2024-01-23 05:36:48
# Regressor Bug: 1833814
# File Overlap Count: 1
# Description:
#   Bug 1833814 - change nsIWidget's cursor logic, r=emilio,win-reviewers,mhowell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D195549
# ==============================================================================

diff -r 94f46aad00df -r 2424d5c30267 widget/windows/nsWindow.cpp
--- a/widget/windows/nsWindow.cpp	Mon Jan 22 20:10:54 2024 +0000
+++ b/widget/windows/nsWindow.cpp	Mon Jan 22 20:42:50 2024 +0000
@@ -3031,7 +3031,10 @@
   sCurrentHCursorIsCustom = false;
   sCurrentCursor = aCursor;
 
-  HCURSOR cursor = CursorForImage(aCursor, GetDefaultScale());
+  HCURSOR cursor = nullptr;
+  if (mCustomCursorAllowed) {
+    cursor = CursorForImage(aCursor, GetDefaultScale());
+  }
   bool custom = false;
   if (cursor) {
     custom = true;
